Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -532,7 +573,19 @@ extension PathHierarchy {
             self.children = [:]
             self.specialBehaviors = []
         }
-        
+
+        /// Initializes a node with a new identifier but the data from an existing node.
+        fileprivate init(
+            cloning source: Node,
+            children: [String: DisambiguationContainer]? = nil
+        ) {
+            self.symbol = source.symbol
+            self.name = source.name
+            self.children = children ?? source.children
```

Comment:
Would it be possible to perform the deep copy recursively in this initializer instead? I believe that would be easier to follow and it would support more than one level of copying.

Reply:
I would still need to filter out the nodes that don't match up with the current symbol graph, which isn't accessible in this initializer. I also can't remove children after the fact since `children`'s setter is marked `private`. (Unless i add a `remove` method, i suppose.)

Reply:
I thought you'd still be able to filter the list before assigning it but I haven't tried. Anyway, this is all private code so we can revisit it later.

Reply:
It would be possible to add a filtering component to the `init(cloning:)` initializer, but then the filtering check would be moved up into the initializer call. I did wind up tweaking the code layout to perform some of the cleanup in that call, but i feel like splitting the loop and the graph inclusion check would be Too Weird for me. Not sure. Maybe there's a better way to do it than what i'm thinking about.